dashboard/app/controllers/datablock_storage_controller.rb

```diff
@@ -0,0 +1,323 @@
+# DatablockStorage is the backend for the 'Data' tab of Applab, the 'Data' blocks
+# and the dataset browser. DatablockStorage stores student-controlled data generated
+# by and with the data blocks like `createRecord`, `updateRecord`, `setKeyValue`, etc.
+#
+# These backend methods are accessed by a thin wrapper on the JS side: datablockStorage.js
+#
+# Student data is stored in corresponding ActiveRecord models:
+# - DatablockStorageTable: stores a list of tables and their columns
+# - DatablockStorageRecord: stores the records in each table
+# - DatablockStorageKvp: stores key-value pairs
+#
+# Metadata for the code.org defined datasets are stored in:
+# - DatablockStorageLibraryManifest
+
+class DatablockStorageController < ApplicationController
+  before_action :validate_channel_id
+  before_action :authenticate_user!
+  skip_before_action :verify_authenticity_token
```

Comment:
why is this needed? You likely can fetch the authenticity token on the frontend

Reply:
My understanding is that since we cache script level pages in CloudFront, the CSRF token we would get from the frontend might be invalid, so we can't rely on it. More [context from slack conversation](https://codedotorg.slack.com/archives/C046G4TRLEN/p1708484595942659?thread_ts=1708478288.327859&cid=C046G4TRLEN)

Reply:
We have created ways of fetching an authenticity token on cached pages, see https://github.com/code-dot-org/code-dot-org/pull/51439, which might be worth a look for so many new routes.

Reply:
Yay! I looked so hard for an equivalent of this before evaluating whether we strictly speaking needed it. Given the number of routes, I agree there's a lot of space for bugs to hide. We've added a followup issue to address this: https://github.com/code-dot-org/code-dot-org/issues/57430